trace/execution_state.go

```diff
@@ -246,6 +246,9 @@ func (state *WorkflowExecutionState) Update(event *history.HistoryEvent) {
 		state.StartTime = event.EventTime
 		state.Attempt = attrs.GetAttempt()
 		state.Type = attrs.GetWorkflowType()
+		if state.Execution.GetRunId() == "" {
+			state.Execution.RunId = attrs.GetOriginalExecutionRunId()
```

Comment:
Hmm.. that seems wrong at first glance, when will you not have the run ID?

Reply:
To accurately get the run ID from history events, you'll need to scan all events in reverse order and take the first one that matches [this](https://github.com/temporalio/api/blob/e7b6fa681f663723b9e419f5fa350e68a3fdfc5c/temporal/api/history/v1/message.proto#L230) or fallback to original run ID.

Reply:
It's the best we can get at this stage of the execution history. This will do for now.

Reply:
I'll open an issue to track and get the correct run Id.

Reply:
Much appreciated! Would you let me know when this happens to learn about it? Thanks